editors/vscode/tests/workspace_folders.spec.ts

```diff
@@ -0,0 +1,36 @@
+import { strictEqual } from "assert";
+import { DiagnosticSeverity, Uri, workspace } from "vscode";
+import { activateExtension, getDiagnostics, loadFixture, sleep, testMultiFolderMode, WORKSPACE_DIR } from "./test-helpers";
+import assert = require("assert");
+
+suiteSetup(async () => {
+  await activateExtension();
+});
+
+const FIXTURES_URI = Uri.joinPath(WORKSPACE_DIR, '..', 'fixtures');
+
+suite('Workspace Folders', () => {
+  testMultiFolderMode('shows diagnostics to newly adding folder', async () => {
+    await loadFixture('debugger');
+    const folderDiagnostics = await getDiagnostics('debugger.js');
+
+    assert(typeof folderDiagnostics[0].code == 'object');
+    strictEqual(folderDiagnostics[0].code.target.authority, 'oxc.rs');
+    strictEqual(folderDiagnostics[0].severity, DiagnosticSeverity.Warning);
+
+    workspace.updateWorkspaceFolders(workspace.workspaceFolders?.length ?? 0, 0, {
```

Comment:
Adding folders only works when we have already a multi folder setup. 
Adding a folder in single folder setup will restart the VSCode Window and makes some weird tests bugs